bin/main

```diff
@@ -1,175 +1,61 @@
-      end
-
-      if draw?(board,win_combinations)
-         puts "#{name1},#{name2} Game has been a Draw Well Played "
-      end    
-  end
-
-  def turn_count(board)
-      counter = 0
-      board.each do |check|
-          if check == "X" || check == "O"
-              counter += 1
-          end
-      end
-    counter
-  end
-
-  def current_player(board,player)
-      if turn_count(board) % 2 == 0 
-          player = "X"
-      else
-          player = "O"
-      end
+    puts 'WELCOME TO TIC TAC TOE YOU WILL LOVE IT'
+    puts "---------------------------------------\n\n"
+    print 'Player 1 name '
+    @name1 = gets.chomp.capitalize!
+    puts ' '
+    print 'Player 2 name '
+    @name2 = gets.chomp.capitalize!
+    puts ' '
+    puts '----------------------------------------------------------------------'
+    puts "|#{@name1} and #{@name2}. You Both are going to be oponent for this Game|"
+    puts '----------------------------------------------------------------------'
+    puts ' '
+    puts 'Below is the Game board you can type your turn that is Specified in the Board'
+    puts ' '
   end
 
   def display_board(board)
-      puts " #{board[0]} | #{board[1]} | #{board[2]} "
-      puts "-----------"
-      puts " #{board[3]} | #{board[4]} | #{board[5]} "
-      puts "-----------"
-      puts " #{board[6]} | #{board[7]} | #{board[8]} "
-      puts " "
-  end
-
-  def name_turn(board,player,name1,name2)
-      if current_player(board,player) == "X"
-          name1
-      else
-          name2
-      end
-  end
-
-  def turn(board,player,name1,name2,win_combinations)
-     puts "Its #{ name_turn(board,player,name1,name2)} turn, your Token is #{current_player(board,player)}"
-     puts "enter no to take position 1-9"
-     input = gets.strip
-     index = input_to_index(input)
-     if valid_move?(board,index)
-         move(board,index,player)
-         display_board(board)
-      else
-          puts "invalid move Type again"
-      end
-  end
-
-  def full?(board)
-    board.all? do |check|
-      if check == "X" || check =="O"
-         true
-      else
-         false
-      end
-    end
+    puts " #{board[0]} | #{board[1]} | #{board[2]} "
+    puts '-----------'
+    puts " #{board[3]} | #{board[4]} | #{board[5]} "
+    puts '-----------'
+    puts " #{board[6]} | #{board[7]} | #{board[8]} "
+    puts ' '
+  end
+
+  def turn(game_board, player, name1, name2, _win_combinations, tic_tac_game)
+    puts "Its #{player.name_turn(game_board.board, player, name1, name2)} turn, your Token is #{player.current_player(game_board.board, player)}"
+    puts 'enter no to take position 1-9'
+    input = gets.strip
+    index = game_board.input_to_index(input)
+    if game_board.valid_move?(game_board.board, index)
+      game_board.move(game_board.board, index, player)
+      tic_tac_game.display_board(game_board.board)
+    else
+      puts 'invalid move Type again'
+     end
   end
```

Comment:
`line 42` is always going to be flagged by rubocop. You just have to learn to write cleaner code. Please let's do the right thing.